packages/pwa-kit-runtime/src/utils/logger.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+const DEFAULT_LOG_LEVEL = 'info'
+
+const isServerSide = typeof window === 'undefined'
+
+const LOG_LEVEL = isServerSide
+    ? process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL
+    : DEFAULT_LOG_LEVEL
+
+export class PWAKITLogger {
+    constructor(logLevel = LOG_LEVEL) {
+        this.logLevel = LOG_LEVELS.includes(logLevel) ? logLevel : DEFAULT_LOG_LEVEL
+    }
```

Comment:
How about allowing users to pass in the `key` to this constructor? That way, we can avoid passing in the `key` on every call to `logger`. 

Perhaps we can have a pattern like this for every package:
```js
// In retail-react-app package
import {PWAKITLogger} from '@salesforce/pwa-kit-runtime/utils/logger'
const logger = new PWAKITLogger({logLevel: 'info', key: 'retail-react-app'})
logger.warn('warn test message', {details: ['App']})

```

Reply:
I like the idea. I implemented the change, so now users can pass the `packageName` when creating the logger. 

The log level and the log format are still something that we want to control globally via an environment variable.